src/preact/component/intersection-observer-resize.js

```diff
@@ -0,0 +1,30 @@
+import {
+  observeWithSharedInOb,
+  unobserveWithSharedInOb,
+} from '#core/dom/layout/viewport-observer';
+
+import {useCallback} from '#preact';
+
+/**
+ * Uses a shared IntersectionObserver per window instance to observe the given `ref`.
+ *
+ * @param {function(IntersectionObserverEntry)} callback
+ * @return {function}
+ */
+export function useIntersectionObserver(callback) {
+  const refCb = useCallback(
+    (node) => {
+      if (!node) {
```

Comment:
When the ref is cleaned up, this will be called with `null`. Which means we'll never unobserve the original node.

Reply:
hah, i said the same ;) https://github.com/ampproject/amphtml/pull/35835#discussion_r700545674

Reply:
Done